tests/complexity.js

```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+var krabbyTest = require('./_baseTest');
+var escomplex = require('escomplex-js');
+var _ = require('lodash');
+var Promise = require('bluebird');
+var fs = Promise.promisifyAll(require("fs"));
+
+var baseConfig = {
+  testConfig: {
+  },
+  files: ['./**/*.js', '!./node_modules/**/*.js', '!./**/node_modules/**/*.js']
+};
+
+var ComplexityTest = new krabbyTest(baseConfig);
+
+ComplexityTest.prototype.analyze = function(files) {
+  var self = this;
+
+  return Promise.map(files, function(fileName) {
+    return fs.readFileAsync(fileName)
+      .then(function(data) {
+        var result = escomplex.analyse(data.toString(), self.config.testConfig);
+        // TODO: Do I even need to do this if the result is already passed all the way to the reporters
+        self.logs.logs.push(result);
+        return result;
+      })
+      .catch(function(e) {
+        // TODO: Why does esprima choke on ./index.js? It might have to do with `#!/usr/bin/env node`
+        // TODO: Where do these get logged?
+        console.warn(fileName, e);
+      });
+  });
+};
+
+ComplexityTest.prototype.process = function(results) {
+  var MAXIMUM_COMPLEXITY = 171;
+  results = _.reject(results, _.isEmpty);
+
+  var complexitySum = _.sum(results, function(result) {
+    // Maintainability: logarithmic scale from -infinity to 171, calculated from the logical lines
+    // of code, the cyclomatix complexity and the Halstead effort. Higher is better.
+    // http://ieeexplore.ieee.org/xpl/login.jsp?tp=&arnumber=242525&url=http%3A%2F%2Fieeexplore.ieee.org%2Fiel2%2F435%2F6237%2F00242525.pdf%3Farnumber%3D242525
+    return result.maintainability;
+  });
+
+  // TODO: Determine if this is an appropriate scoring method for logarithmic data
+  results.grade = complexitySum / results.length / MAXIMUM_COMPLEXITY;
+  return results;
+};
+
+ComplexityTest.prototype.test = function(cb) {
+  this.analyze(this.config.files)
+    .then(this.process)
+    .nodeify(cb);
```

Comment:
I could put one as a general catch for my process method but I don't expect that to throw and there is already a specific catch for the analyze fn.